File: NET/NumberGuess.cs
using System;
using System.Collections.Generic;

namespace TSAF{
    public class NumberGuess{
        public static void Main(){
            Console.Write("What is your name? ");
            string name = Console.ReadLine();
            Console.WriteLine("Hello " + name + ", I want to play a game.");

            Random random = new Random();
            int number = random.Next(1, 100);
            int attempts = 0;
            List<int> guesses = new List<int>();
            int guess = 0;
            
            while(attempts < 10){
                try{
                    Console.Write("Guess a number: ");
                    guess = int.Parse(Console.ReadLine());
                    guesses.Add(guess);
                } 
                catch(Exception){
                    Console.WriteLine("You entered a non valid integer!");
                    continue;
                }

                attempts++;

                if(guess > number){
                    Console.WriteLine("Your guess is too high!");
                }
                else if(guess < number){
                    Console.WriteLine("Your guess is too low!");
                }
                else{
                    Console.WriteLine("That is correct!");
                    Console.WriteLine("Found answer in attempt " + attempts);
                    break;
                }
                Console.WriteLine("Attempts made: " +  attempts);
            }

            Console.WriteLine("Guess history:");
            foreach(int value in guesses){
                Console.WriteLine(value);
            }

            Console.WriteLine("Game over!");
        }
    }
}

File: C-CPP/NumberGuess.cpp
#include <stdio.h>
#include <stdlib.h>
#include <time.h>
#include <ctype.h>
#include <iostream>
#include <limits>

int main(void) {
  char name[50];
  printf("What is your name? ");
  scanf("%s", name);
  printf("Hello %s, I want to play a game.\n", name);

  srand(time(NULL));
  int number = rand() % 100 + 1;
  int turns = 10;
  int attempts = 0;
  int guess = 0;
  int guesses[turns];

  while(attempts < turns){
    try{
      printf("Guess a number: ");
      if(!(std::cin >> guess)){
          std::cin.clear();
          std::cin.ignore(std::numeric_limits<std::streamsize>::max(), '\n');
          throw -1;
      }
    }
    catch(int n){
      printf("You entered a non valid integer!\n");
      continue;
    }
    guesses[attempts] = guess;
    attempts += 1;

    if(guess > number){
      printf("Your guess is too high!\n");
    }
    else if(guess < number){
      printf("Your guess is too low!\n");
    }
    else{
      printf("That is correct!\n");
      printf("Found answer in attempt %d\n", attempts);
      break;
    }

    printf("Attemps made: %d\n", attempts);
  }

  printf("Guess history:\n");
  for(int i = 0; i < attempts; i++){
    printf("%d\n", guesses[i]);
  }
}